### ISSUE TITLE
Wanna to get gradients and hessians

### ISSUE THREAD

**Issue by skb888** [dmlc/xgboost#4818]
For the Federated XGBoost, the local clients pass g and h (gradients and hessians) to the master server. Is there a way to gain g and h by using XGBoost API? Here, g and h are values needed during the iteration through a sorted feature array (greedy or quantile). It seems that even if the core.py includes update function, the g and h values are computed via calling _LIB.XGBoosterUpdateOneIter(). While, this LIB is written in cpp. So is there any way can get g and h by modifying some python codes. Thanks

**Comment by trivialfis** [comment]
@skb888 It's computed in objectives.  So provide your own custom objective then you can calculate gradients yourself.

**Comment by trivialfis** [comment]
We have a tutorial in master branch doc/ for doing just that.  See https://xgboost.readthedocs.io/en/latest/tutorials/custom_metric_obj.html .

**Comment by hcho3** [comment]
@trivialfis I think the OP is asking a way to obtain gradient/hessian values that are being computed by `XGBoosterUpdateOneIter()`. @skb888 Unfortunately, it is currently not possible to do this in Python.

**Comment by RAMitchell** [comment]
I would just reimplement the desired loss function via python, then you would have access to it.
